File: src/main/java/Card.java
package main.java;

public class Card {

    public int getCardValue() {
        return cardValue.value;
    }

    public boolean isAce(){
        return cardValue.name().equalsIgnoreCase("ACE");
    }

    
    public String toString() {
        return cardSuit.toString() + cardValue.toString();
    }

    private final Value cardValue;
    private final Suit cardSuit;

    public Card(Value cardValue, Suit cardSuit) {
        this.cardValue = cardValue;
        this.cardSuit = cardSuit;
    }

    enum Value {

        TWO(2), THREE(3), FOUR(4), FIVE(5), SIX(6), SEVEN(7),
        EIGHT(8), NINE(9), TEN(10), JACK(10), QUEEN(10), KING(10), ACE(11);

        private final int value;
        Value(int value) {
            this.value = value;
        }

        @Override
        public String toString() {
            return switch (this){
                case ACE -> "A";
                case KING -> "K";
                case QUEEN -> "Q";
                case JACK -> "J";
                case TEN -> "10";
                case NINE -> "9";
                case EIGHT -> "8";
                case SEVEN -> "7";
                case SIX -> "6";
                case FIVE -> "5";
                case FOUR -> "4";
                case THREE -> "3";
                case TWO -> "2";
            };
        }
    }
    enum Suit {
        CLUBS, HEARTS, DIAMONDS, SPADES;

        @Override
        public String toString() {
            return switch (this){
                case CLUBS -> "C";
                case HEARTS -> "H";
                case DIAMONDS -> "D";
                case SPADES -> "S";
            };
        }
    }
}

File: src/main/kotlin/Card.kt
package main.kotlin

class Card(private val cardValue: Value, private val cardSuit: Suit) {

    fun getCardValue(): Int {
        return cardValue.value
    }

    val isAce: Boolean
        get() = cardValue.name.equals("ACE", ignoreCase = true)

    override fun toString(): String {
        return cardSuit.toString() + cardValue.toString()
    }

    enum class Value(val value: Int) {
        TWO(2), THREE(3), FOUR(4), FIVE(5), SIX(6), SEVEN(7), EIGHT(8),
        NINE(9), TEN(10), JACK(10), QUEEN(10), KING(10), ACE(11);

        override fun toString(): String {
            return when (this) {
                ACE -> "A"
                KING -> "K"
                QUEEN -> "Q"
                JACK -> "J"
                TEN -> "10"
                NINE -> "9"
                EIGHT -> "8"
                SEVEN -> "7"
                SIX -> "6"
                FIVE -> "5"
                FOUR -> "4"
                THREE -> "3"
                TWO -> "2"
            }
        }
    }

    enum class Suit {
        CLUBS, HEARTS, DIAMONDS, SPADES;

        override fun toString(): String {
            return when (this) {
                CLUBS -> "C"
                HEARTS -> "H"
                DIAMONDS -> "D"
                SPADES -> "S"
            }
        }
    }
}